tests/s25Main/mapGenerator/testHeadQuarters.cpp

```diff
@@ -174,10 +217,13 @@ BOOST_AUTO_TEST_CASE(PlaceHeadQuarters_returns_true_for_any_player_number_on_sui
     {
         RunTest(size, [&size, &rnd, players](Map& map, TextureMap& textures) {
             map.textures.Resize(size, TexturePair(textures.Find(IsBuildableLand)));
-
-            auto success = PlaceHeadQuarters(map, rnd, players);
-
-            BOOST_REQUIRE(success);
+            try
+            {
+                PlaceHeadQuarters(map, rnd, players);
+            } catch(...)
+            {
+                BOOST_FAIL("Failed to place HQs");
```

Comment:
No need for this, a test would fail if an exception is uncaught. If you want you can use `BOOST_REQUIRE_NO_THROW` to be explicit